sfepy/terms/terms_surface.py

```diff
@@ -95,6 +101,86 @@ def set_arg_types( self ):
         else:
             self.function = self.d_fun
 
+
+class SDLinearTractionTerm(Term):
+    r"""
+    Sensitivity of the linear traction term.
+
+    :Definition:
+
+    .. math::
+        \int_{\Gamma} \ul{v} \cdot \ull{\sigma} \cdot \ul{n},
+        \int_{\Gamma} \ul{v} \cdot \ul{n},
+
+    :Arguments:
+        - material  : :math:`\ull{\sigma}`
+        - parameter : :math:`\ul{v}`
+    """
+
+    name = 'd_sd_surface_ltr'
+    arg_types = ('opt_material', 'parameter', 'parameter_mv')
+    arg_shapes = [{'opt_material': 'S, 1', 'parameter': 'D',
+                   'parameter_mv': 'D'}, {'opt_material': '1, 1'},
+                  {'opt_material': 'D, 1'}, {'opt_material': 'D, D'},
+                  {'opt_material': None}]
+    integration = 'surface'
+
+    @staticmethod
+    def d_fun(out, traction, val, grad_mv, div_mv, sg):
+
+        tdim, tdim2 = (None, None) if traction is None else traction.shape[2:]
+        dim = val.shape[2]
+        sym = (dim + 1) * dim // 2
+        nel, nqp = div_mv.shape[:2]
```

Comment:
In `get_eval_shape()` there is `n_el`, `n_qp`...

Reply:
`SDLinearTractionTerm.d_fun()` is the static method so I would have to call `div_mv.get_data_shape()`, but it requires `integrals`, etc. I think the current solution is simple and reliable.

Reply:
Yes, just that you use different names for the same things (nel x n_el, ...)

Reply:
renamed: `nel`, `nqp` --> `n_el`, `n_qp`